scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -538,73 +595,66 @@ EnhancedVolcano::EnhancedVolcano(deseq_results,
 
 ```{r}
 # filter to just myoblast cells and remove any NA's before plotting
 myoblast_combined_sce <- rms_sce[, which(rms_sce$celltype_broad == "Tumor_Myoblast")]
 
-# first look at just ARMS vs. ERMS 
+# plot PTPRT expression in ARMS vs. ERMS
 scater::plotReducedDim(myoblast_combined_sce,
                        dimred = "fastmnn_UMAP",
                        colour_by = "ENSG00000196090", #PTPRT
                        point_size= 0.5,
                        point_alpha = 0.4,
                        other_fields = "diagnosis_group") +
-  facet_wrap(~ diagnosis_group, nrow = 3)
+  facet_grid(cols = vars(diagnosis_group))
 ```
 
+In the above plot we only plotted the tumor myoblast cells that we used in our DE analysis. 
+However, we might be interested to see the expression of genes that are differentially expressed in tumor myoblasts in other cell types present in our samples.
 
 ```{r}
-# let's look across some other cell types
+# let's compare gene expression across some other cell types
 celltypes <- c("Tumor_Myoblast", "Tumor_Mesoderm", "Tumor_Myocyte", "Vascular Endothelium")
 
 # subset to just tumor celltypes that we are interested in
 tumor_sce <- rms_sce[, which(rms_sce$celltype_broad %in% celltypes)]
 ```
 
+Next we will look at a few DE genes that we identified, one upregulated gene and one downregulated gene, and compare their expression in myoblasts to other cell types in ARMS and ERMS samples.
+We will use the `scater::plotExpression()` function to create a violin plot with RMS subtype on the x-axis and gene expression on the y-axis.
+We can continue using our `facet_grid()` functionality and show each cell type in its own column of the plot grid.
+Because we want to show multiple genes here, we are going to add an additional option to `facet_grid()` to include multiple rows in our plot grid, one for each gene of interest.
 
 ```{r}
 # pick a few genes to look at 
 genes_to_plot <- c("ENSG00000196090", #PTPRT
                    "ENSG00000148935") #GAS2
 
-# create a violin plot across cell types of interest
+# create a violin plot 
 scater::plotExpression(tumor_sce,
-                       features = genes_to_plot, #PTPRT
+                       features = genes_to_plot, # provide a vector of genes to plot
                        x = "diagnosis_group", 
                        colour_by = "diagnosis_group",
                        other_fields = "celltype_broad",
                        point_size = 0.1) +
-  facet_grid(Feature ~ celltype_broad) + 
-  theme(strip.text = element_text(size = 7)) + 
-  guides(colour = guide_legend(override.aes = list(size = 3, alpha = 1)))
+  facet_grid(cols = vars(celltype_broad), # each celltype is its own column
+             rows = vars(Feature)) + # each feature is its own row
+  theme(strip.text = element_text(size = 7)) + # change the font size of the facet labels
+  guides(colour = guide_legend(override.aes = list(size = 3, alpha = 1))) # change the size of the legend colors
```

Comment:
As long as we are getting into it, do we want to update the legend title?
```suggestion
  guides(colour = guide_legend(
      title = "Subtype", # update the legend title
     override.aes = list(size = 3, alpha = 1)) # change the size of the legend colors
 ) 
```

Reply:
Noting that the long comments here might be too long, so we might move them to above the lines instead of at the end.